sdm-api-thermostat.groovy

```diff
@@ -95,30 +95,46 @@ def off() {
 }
 
 def setCoolingSetpoint(temp) {
+    def tempMovement = device.currentValue('heatingSetpoint').toFloat() - temp.toFloat() + 1.5
+    if (tempMovement <= 0) {
+        tempMovement = device.currentValue('heatingSetpoint')
+    } else {
+        tempMovement = device.currentValue('heatingSetpoint').toFloat() - tempMovement.toFloat()
+    }
     def mode = device.currentValue('thermostatMode')
     if (mode == 'cool') {
         parent.deviceSetTemperatureSetpoint(device, null, temp)
     } else if (mode == 'auto') {
-        parent.deviceSetTemperatureSetpoint(device, device.currentValue('heatingSetpoint'), temp)
+        parent.deviceSetTemperatureSetpoint(device, tempMovement, temp)
     } else {
         log.warn("Cannot setCoolingSetpoint in thermostatMode: ${mode}")
     }
 }
 
 def setHeatingSetpoint(temp) {
+    def tempMovement = temp.toFloat() - device.currentValue('coolingSetpoint').toFloat() + 1.5
+    if (tempMovement <= 0) {
+        tempMovement = device.currentValue('coolingSetpoint')
+    } else {
+        tempMovement = device.currentValue('coolingSetpoint').toFloat() + tempMovement.toFloat()
+    }
     def mode = device.currentValue('thermostatMode')
     if (mode == 'heat') {
         parent.deviceSetTemperatureSetpoint(device, temp, null)
     } else if (mode == 'auto') {
-        parent.deviceSetTemperatureSetpoint(device, temp, device.currentValue('coolingSetpoint'))
+        parent.deviceSetTemperatureSetpoint(device, temp, tempMovement)
     } else {
         log.warn("Cannot setHeatingSetpoint in thermostatMode: ${mode}")
     }
 }
 
 def setHeatCoolSetpoint(heat, cool) {
+    def tempMovement = heat.toFloat() - cool.toFloat() + 1.5
+    if (tempMovement < 0) {
```

Comment:
do we need this? check below is (tempMovement <= 0)

Reply:
No need for this.  Cut/paste issue from the previous functions.